test/test_rnn.lua

```diff
@@ -0,0 +1,178 @@
+require 'cudnn'
+require 'cunn'
+local ffi = require 'ffi'
+local errcheck = cudnn.errcheck
+
+local cudnntest = torch.TestSuite()
+local mytester
+
+local tolerance = 70
+
+function cudnntest.testRNN()
+    local miniBatch = 64
+    local seqLength = 20
+    local hiddenSize = 512
+    local numberOfLayers = 2
+    local numberOfLinearLayers = 2
+    local checkSums = getRNNCheckSums(miniBatch, seqLength, hiddenSize, numberOfLayers, numberOfLinearLayers)
+
+    print(string.format("i %E", checkSums.localSumi))
+    print(string.format("h %E", checkSums.localSumh))
+    print(string.format("c %E", checkSums.localSumc))
+    print(string.format("di %E", checkSums.localSumdi))
+    print(string.format("dh %E", checkSums.localSumdh))
+    print(string.format("dw %E", checkSums.localSumdw))
+
+    -- Checksums to check against are retrieved from cudnn RNN sample.
+    mytester:assertalmosteq(checkSums.localSumi, 1.315793E+06, tolerance, 'checkSum with reference for localsumi failed')
+    mytester:assertalmosteq(checkSums.localSumh, 1.315212E+05, tolerance, 'checkSum with reference for localSumh failed')
+    mytester:assertalmosteq(checkSums.localSumc, 0.000000E+00, tolerance, 'checkSum with reference for localSumc failed')
+    mytester:assertalmosteq(checkSums.localSumdi, 6.676003E+01, tolerance, 'checkSum with reference for localSumdi failed')
+    mytester:assertalmosteq(checkSums.localSumdh, 6.425067E+01, tolerance, 'checkSum with reference for localSumdh failed')
+    mytester:assertalmosteq(checkSums.localSumdw, 1.453750E+09, tolerance, 'checkSum with reference for localSumdw failed')
+end
+
+--[[
+-- Method gets Checksums of RNN to compare with ref Checksums in cudnn RNN C sample.
+-- ]]
+function getRNNCheckSums(miniBatch, seqLength, hiddenSize, numberOfLayers, numberOfLinearLayers)
+
+    local input = torch.CudaTensor(seqLength, miniBatch, hiddenSize):fill(1) -- Input initialised to 1s.
+    local rnn = cudnn.RNN(hiddenSize, hiddenSize, numberOfLayers)
+    rnn:resetWeightDescriptor()
+
+    -- Matrices are initialised to 1 / matrixSize, biases to 1.
+    for layer = 0, numberOfLayers - 1 do
+        for layerId = 0, numberOfLinearLayers - 1 do
+            local linLayerMatDesc = rnn:createFilterDescriptors(1)
+            local matrixPointer = ffi.new("float*[1]")
+            errcheck('cudnnGetRNNLinLayerMatrixParams',
+                cudnn.getHandle(),
+                rnn.rnnDesc[0],
+                layer,
+                rnn.xDescs,
+                rnn.wDesc[0],
+                rnn.weight:data(),
+                layerId,
+                linLayerMatDesc[0],
+                ffi.cast("void**", matrixPointer))
+
+            local dataType = 'CUDNN_DATA_FLOAT'
+            local format = 'CUDNN_TENSOR_NCHW'
+            local nbDims = torch.IntTensor(1)
+            local filterDimA = torch.IntTensor({ 1, 1, 1 })
```

Comment:
You can remove hard-coded 3 dimensions by 
```
local minDim=3
local filterDimA=torch.ones(minDim)
```
replace '3' argument to GetFilterNdDescriptor call with ```minDim```, and replace ```filterDimA[1] * filterDimA[2] * filterDimA[3]``` with ```filterDimA:prod()```